src/zenml/integrations/aws/service_connectors/aws_service_connector.py

```diff
@@ -1371,6 +1372,21 @@ def _configure_local_client(
                     "aws_session_token"
                 ] = credentials.token
 
+            aws_credentials_path = os.path.join(
+                users_home, ".aws", "credentials"
+            )
+
+            # Ensure the .aws directory exists
+            os.makedirs(os.path.dirname(aws_credentials_path), exist_ok=True)
+
+            # Ensure the credentials file exists
+            if not os.path.isfile(aws_credentials_path):
+                open(aws_credentials_path, "a").close()
```

Comment:
```suggestion
            # Ensure the credentials file exists
            if not os.path.isfile(aws_credentials_path):
              with open(aws_credentials_path, "a"):
                pass  # The file is automatically closed when exiting the with block
```

A bit more Pythonic...